src/core/ext/filters/client_channel/service_config.h

```diff
@@ -120,9 +149,16 @@ class ServiceConfig : public RefCounted<ServiceConfig> {
       grpc_json* json, CreateValue<T> create_value,
       typename SliceHashTable<RefCountedPtr<T>>::Entry* entries, size_t* idx);
 
+  static constexpr int kMaxParsers = 32;
+  static int registered_parsers_count;
+  static ServiceConfigParser registered_parsers[kMaxParsers];
+
   UniquePtr<char> service_config_json_;
   UniquePtr<char> json_string_;  // Underlying storage for json_tree.
   grpc_json* json_tree_;
+
+  InlinedVector<UniquePtr<ServiceConfigParsedObject>, kMaxParsers>
```

Comment:
I think we'll also need to store a method config table, where the value type of the table is this same type.

Reply:
I was planning to let the method config table be a part of the service config parsed object

Reply:
That would mean that each filter would have its own method config table, which means that each filter would have to independently look up the call's method in their own instance of the table.  It seems like it would be better for performance to have a single method config table whose value contains the per-method data for every filter.  That way, we can do the method config table lookup only once per call, in the client_channel filter.  We would then pass the resulting array of parsed objects into the subchannel stack, so each filter could grab its data via the index of its parser.

Reply:
done!